modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/parser/ParserTestContentProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.hdb.ds.test.parser;
+
+import com.sap.xsk.exceptions.XSKArtifactParserException;
+import com.sap.xsk.hdb.ds.api.XSKDataStructuresException;
+import com.sap.xsk.hdb.ds.model.XSKDataStructureParametersModel;
+import com.sap.xsk.hdb.ds.model.hdbprocedure.XSKDataStructureHDBProcedureModel;
+import com.sap.xsk.hdb.ds.model.hdbtablefunction.XSKDataStructureHDBTableFunctionModel;
+import com.sap.xsk.hdb.ds.parser.XSKDataStructureParser;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+public class ParserTestContentProvider {
+
+    String getTestContent(String location) throws IOException {
+        InputStream in = ParserTestContentProvider.class.getResourceAsStream(location);
+        return IOUtils.toString(Objects.requireNonNull(in), StandardCharsets.UTF_8);
+    }
+
+    XSKDataStructureHDBProcedureModel parseProcedure(@NotNull XSKDataStructureParser<XSKDataStructureHDBProcedureModel> parser, String location, String content)
```

Comment:
Because in this class models for both artifacts are built. Class is used only to build models and used it in the mocked parser.

Reply:
I think it would be better to keep the parsing in the appropriate test class as it would be a bit easier to follow the test logic. Having a class named `ParserTestContentProvider` that also has methods like `parseTableFunction` is a bit strange. Please, move `parseProcedure` and `parseTableFunction` methods in the dedicated classes.